### DISCUSSION TITLE
might be need update docs for Vite setting

### DISCUSSION THREAD

**Discussion by ryota-murakami** [mswjs/msw#1077] [General]
I have a OSS that React + TS [Vite](https://vitejs.dev/) template and added msw setup recently. [vite-react-ts-extended](https://github.com/laststance/vite-react-ts-extended)

I noticed that Vite's dev-server serving React code before msw Service Worker enabled within [general setting](https://mswjs.io/docs/getting-started/integrate/browser#start-worker) like following screen shot(please see console log message orders).  

<img width="727" alt="Screen Shot 2022-01-23 at 0 28 15" src="https://user-images.githubusercontent.com/5501268/150994381-662c4505-15ad-4c76-a873-2ed86cdd6171.png">

So, I added Promise to ensure execute Service Worker registration code firstly, then msw working fine.  

```typescript
const runReact = () => {
  // @ts-ignore v18 @types coming not yet
  const root = ReactDOM.createRoot(document.getElementById('root'))
  root.render(<App />)
}

if (process.env.NODE_ENV === 'development') {
  import('../mocks/browser')
    .then(({ worker }) => {
      worker.start()
    })
    .then(() => runReact())
} else {
  runReact()
}
```

<img width="793" alt="success" src="https://user-images.githubusercontent.com/5501268/150995152-96e30897-d59c-4b84-95d8-c978104f5343.png">

I guess this behavior relevant to [dependency prebuild](https://vitejs.dev/guide/dep-pre-bundling.html#dependency-pre-bundling), there is no evidence though.  

I'm not sure my above solution is ideal one, but I think we have to update for Vite's msw setting documentation.  

- [reproduction here](https://github.com/ryota-murakami/msw-vite-fetch-failed-reproduction)
```
git clone https://github.com/ryota-murakami/msw-vite-fetch-failed-reproduction.git
cd msw-vite-fetch-failed-reproduction
yarn
yarn dev
// then open browser and you can see reproducion in dev tools
```

- [fixed with Promise code here](https://github.com/laststance/vite-react-ts-extended/blob/main/src/main.tsx)

**Comment by kettanaito** [comment]
Hey, @ryota-murakami! Thanks for reaching out. 

We don't currently have a Vite example in our [examples](https://github.com/mswjs/examples/tree/2338457a82f60e5d76a9a32a742157af21bf3606/examples). We do mention [Deferred mounting](https://mswjs.io/docs/recipes/deferred-mounting) recipe in the documentation for anybody who's faced with race conditions between their app making a request and the service worker registering. This looks exactly like what's happening in your case. I'm not sure if there's anything to update.

**Reply by ryota-murakami** [reply]
@ryota-murakami Thank you for speedy response!
I missed [Deferred mounting](https://mswjs.io/docs/recipes/deferred-mounting), that completely covering Vite usecase!
I think current document is writing enough and we don't have to adding any extra section.

Thank you so much! 😄
